src/templates/FacilityEdit.vue

```diff
@@ -0,0 +1,222 @@
+          <b-form-group id="entity-name" label="Name">
+            <b-form-input type="text" required v-model="entity.name" />
+          </b-form-group>
+          <b-form-group id="entity-address" label="Address">
+            <b-form-input type="text" v-model="entity.address.street[0]" placeholder="Street" />
+            <b-form-input type="text" v-model="entity.address.city" placeholder="City" />
+            <b-form-select v-model="stateSelected" :options="stateOptions"></b-form-select>
+            <b-form-input type="text" v-model="entity.address.zip" placeholder="Zip" />
+          </b-form-group>
+          <b-form-group id="entity-type" label="Type">
+            <b-form-select v-model="typeSelected" required :options="typeOptions"></b-form-select>
+          </b-form-group>
+          <b-form-group id="entity-notes" label="Notes">
+            <b-form-input type="text" v-model="entity.attributes.notes" />
+          </b-form-group>
+          <b-button
+            type="submit"
+            variant="primary"
+          >{{ this.$route.params.entityID ? 'Update facility' : 'Create facility' }}</b-button>
+          <b-button
+            type="cancel"
+            variant="outline-secondary"
+            @click.prevent="returnToLastPage"
+          >Cancel</b-button>
+        </b-form>
+      </b-col>
+    </b-row>
+    <br />
+
+    <b-row v-if="$route.params.entityID">
+      <b-col cols="6">
+        <h4>Linked Contacts</h4>
+        <b-input-group v-if="entity.contacts.length > 0">
+          <b-form-select v-model="selectedContactID" :options="contactSelectList"></b-form-select>
+          <b-button
+            type="submit"
+            variant="primary"
+            @click.prevent="unlinkEntity"
+          >Unlink from Facility</b-button>
+        </b-input-group>
+        <p v-if="entity.contacts.length === 0">This facility has no linked contact.</p>
+        <br />
+        <div v-if="showAdmin">
+          <h4>Admin Options</h4>
+          <b-button type="submit" variant="primary" @click.prevent="deleteFacility">Delete Facility</b-button>
+        </div>
+      </b-col>
+    </b-row>
+  </b-container>
+</template>
+
+<script>
+export default {
+  name: "Entity",
+  data() {
+    const showAdmin =
+      this.$jwt.decode(this.$root.auth_token).type === "admin" ? true : false
+    return (
+      {
+        entity: {
+          id: null,
+          name: null,
+          address: {
+            street: [""],
+            city: "",
+            state: "",
+            zip: ""
+          },
+          attributes: {
+            notes: ""
+          },
+          contacts: []
+        },
+        showAdmin,
+        contactSelectList: [{ value: null, text: "Contacts Linked" }],
+        selectedContactID: null,
+        stateSelected: "MD",
+        stateOptions: [{ value: "MD", text: "Maryland" }],
+        typeSelected: null,
+        typeOptions: [
+          { value: null, text: "Select facility type" },
```

Comment:
I think you could clean this array generation up a little, but not a big deal

Reply:
Got rid of test, but this is a placeholder until we add a type endpoint to the API.

Reply:
https://github.com/CodeForBaltimore/Bmore-Responsive/issues/219